service/topology_coordinator.cc

```diff
@@ -1126,6 +1126,8 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
     }
 
     void generate_migration_update(std::vector<canonical_mutation>& out, const group0_guard& guard, const tablet_migration_info& mig) {
+        SCYLLA_ASSERT(get_token_metadata_ptr()->tablets().is_base_table(mig.tablet.table));
```

Comment:
Please resist the urge to use SCYLLA_ASSERT - see https://github.com/scylladb/scylladb/issues/7871
We should be using on_internal_error() or `utils::on_internal_error`, not SCYLLA_ASSERT.
Perhaps we need convenience wrappers for on_internal_error(), but it's not an excuse to use SCYLLA_ASSERT.